### PR TITLE
Contributing Member use case

### PR DESCRIPTION
**Pull request by ThorbenStaufer**
Added a new use case for receiving a free membership, if the user is a valuable community member.

## Basics

- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is conforming to [our Documentation Guidelines](/doc/documentation.md)
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to our Coding Guidelines
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

### CODE REVIEW

**Review by absurd-turtle — APPROVED** [review]

**Review by Benjaminimal — COMMENTED** [review]
What is a membership?

**Comment by markus2330** [comment]
This should be explained in the glossary :wink:

File: doc/usecases/gain_free_membership.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Use Case: Gain Free Membership
+
+## Summary
+
+- **Scope:** Membership
+- **Level:** User Goal
+- **Actors:** App User
+- **Brief:** The user receives a free membership through using the app.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:8]:
```suggestion
- **Brief:** The user receives a free membership through contributions.
```

File: doc/usecases/gain_free_membership.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- **Status:** Draft
+
+## Scenarios
+
+- **Precondition:**
+  - The user already has an existing membership OR
+  - The user has currently no membership.
+- **Main success scenario:**
+  - The user can receive PermaplanT points through various activities in the app, like:
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:17]:
```suggestion
  - The user can receive permacoins through various activities in the app, like:
```

File: doc/usecases/gain_free_membership.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    - gaining [Blossoms](./gamification/gain_blossoms.md) in the Expert Track.
+    - getting likes from other users on their reviews.
+    - getting [approvals](./polyculture/review_plant_relationships.md) on their submitted [plant relationships](./polyculture/add_plant_relationships.md).
+    - getting their comments on a map marked as helpful by the maps owner.
+  - The user can see their exact count of PermaplanT points in their profile.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:22]:
```suggestion
  - The user can see their exact count of current and previous permacoins in their profile.
```

File: doc/usecases/gain_free_membership.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  - When reaching a certain milestone with the amount of gathered PermaplanT points, the user will be given the current year of membership for free.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:23]:
```suggestion
  - If the user applied for the contribution membership and when reaching a certain milestone with the amount of gathered permacoins, the user will be given the current calender year of membership without compensation via money.
```

File: doc/usecases/gain_free_membership.md
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:1]: Please rename to "contribution_membership", free implies you get it for free, which is not the case...

File: doc/usecases/gain_free_membership.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  - PermaplanT points will be reset to zero at the end of the membership year.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:24]: We remember all permacoins of every year in a protocol (that only gets expanded and never changed). The user starts every year from 0. Please clarify this here.

```suggestion
```

File: doc/usecases/gain_free_membership.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+- **Alternative scenario:**
+  - The user does not gather enough PermaplanT points and has to pay the full price for this years membership.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:26]:
```suggestion
  - The user does not gather enough permacoins and has to pay the full price for this years membership.
```

File: doc/usecases/gain_free_membership.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+- **Precondition:**
+  - The user already has an existing membership OR
+  - The user has currently no membership.
+- **Main success scenario:**
+  - The user can receive PermaplanT points through various activities in the app, like:
+    - gaining [Blossoms](./gamification/gain_blossoms.md) in the Expert Track.
+    - getting likes from other users on their reviews.
+    - getting [approvals](./polyculture/review_plant_relationships.md) on their submitted [plant relationships](./polyculture/add_plant_relationships.md).
+    - getting their comments on a map marked as helpful by the maps owner.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:21]:
```suggestion
    - getting their comments on a map marked as helpful by the maps owner.
    - recruit of new members.
```

File: doc/usecases/gain_free_membership.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  - The user can see their exact count of PermaplanT points in their profile.
+  - When reaching a certain milestone with the amount of gathered PermaplanT points, the user will be given the current year of membership for free.
+  - PermaplanT points will be reset to zero at the end of the membership year.
+- **Alternative scenario:**
+  - The user does not gather enough PermaplanT points and has to pay the full price for this years membership.
+- **Error scenario:**
+  - Due to an internal server error, the user is charged with this years membership even though (s)he gathered enough points.
+    In this case, the user can contact the PermaplanT team via official channels so that an administrator can investigate the eligibility of the user for a free membership.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:29]: This doesn't really add something to the use case, it simply states the obvious:

```suggestion
```

File: doc/usecases/gain_free_membership.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+- **Postcondition:**
+  - The user can use the app for free if (s)he is a valuable member of the community.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:31]: please avoid "free" also here.

File: doc/usecases/gain_free_membership.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+- **Non-functional Constraints:**
+  - The amount of gathered PermaplanT point in each year should be kept in the database for administrative purposes.
```
Comment (markus2330) [inline doc/usecases/gain_free_membership.md:33]: Should already be clear above, so probably not needed here.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Thank you.

File: doc/usecases/contributing_member.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    - getting their comments on a map marked as helpful by the maps owner.
+    - recruiting new members.
+  - The user can see their exact count of current and previous permacoins in their profile.
+  - If the user applied for the contributing membership and when reaching a certain milestone with the amount of gathered permacoins, the user will be given the current calendar year of membership without compensation via money.
+  - Permacoins can not be transferred from the current into the next year.
+- **Alternative scenario:**
+  - The user does not gather enough permacoins and has to pay the full price for this years membership.
+- **Error scenario:**
+- **Postcondition:**
+  - The user can use the app without compensation via money if (s)he is a valuable member of the community.
```
Comment (markus2330) [inline doc/usecases/contributing_member.md:30]:
```suggestion
  - The user has the permission to use the app and to be part of the community.
```

File: doc/usecases/contributing_member.md [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+- **Main success scenario:**
+  - The user can receive permacoins through various activities in the app, like:
+    - gaining [Blossoms](./gamification/gain_blossoms.md) in the Expert Track.
+    - getting likes from other users on their reviews.
+    - getting [approvals](./polyculture/review_plant_relationships.md) on their submitted [plant relationships](./polyculture/add_plant_relationships.md).
+    - getting their comments on a map marked as helpful by the maps owner.
+    - recruiting new members.
+  - The user can see their exact count of current and previous permacoins in their profile.
+  - If the user applied for the contributing membership and when reaching a certain milestone with the amount of gathered permacoins, the user will be given the current calendar year of membership without compensation via money.
+  - Permacoins can not be transferred from the current into the next year.
```
Comment (markus2330) [inline doc/usecases/contributing_member.md:25]:
```suggestion
  - Permacoins can not be transferred from the current into another year nor be exchanged to money in any way.
```

**Review by markus2330 — APPROVED** [review]
Great job, two last minor fixes, then it can be merged (by someone else).

**Comment by markus2330** [comment]
jenkins build please

**Comment by markus2330** [comment]
Now part of #366

**Pull request closed without merging**